tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -115,23 +116,35 @@ initialize_column_state(DecompressChunkState *state)
 	ScanState *ss = (ScanState *) state;
 	TupleDesc desc = ss->ss_ScanTupleSlot->tts_tupleDescriptor;
 	ListCell *lc;
-	int i;
 
-	state->num_columns = list_length(state->varattno_map);
+	if (list_length(state->decompression_map) == 0)
+	{
+		elog(ERROR, "no columns specified to decompress");
+	}
 
-	state->columns = palloc0(state->num_columns * sizeof(DecompressChunkColumnState));
+	state->columns =
+		palloc0(list_length(state->decompression_map) * sizeof(DecompressChunkColumnState));
```

Comment:
Do you want to remove `num_columns` from the state? Lists have `length` now, so it is fast to compute and keeping duplicate information just increase the risk of bugs.

Reply:
I would rather remove the `decompression_map` because it's used only for initialization, and keep the array + length pair, it's idiomatic. But there's no convenient way to do that, we only have the single executor state struct that holds both the initialized state and the input for initialization.

Reply:
Ok. Let's consider what we can do here later then, but might be a good idea to add a few asserts to make sure that no bugs creep in.